src/backend/commands/createas.c

```diff
@@ -348,6 +348,8 @@ ExecCreateTableAs(CreateTableAsStmt *stmt, const char *queryString,
 			ParseState *pstate = make_parsestate(NULL);
 			FuncCall *fn;
 
+			pstate->p_expr_kind = EXPR_KIND_SELECT_TARGET;
+
```

Comment:
It seems good to me. 

We might be able to make a function to initialize pstate for our purpose, but I think this fix is enough for now.